src/deploy-cloudrun.ts

```diff
@@ -31,144 +60,160 @@ export async function run(): Promise<void> {
   core.exportVariable(GCLOUD_METRICS_ENV_VAR, GCLOUD_METRICS_LABEL);
   try {
     // Get inputs
-    // Core inputs
-    const image = core.getInput('image'); // Image ie gcr.io/...
-    const name = core.getInput('service'); // Service name
-    const metadata = core.getInput('metadata'); // YAML file
-    const credentials = core.getInput('credentials'); // Service account key
-    let projectId = core.getInput('project_id');
-    let gcloudVersion = core.getInput('gcloud_version');
-    // Flags
-    const envVars = core.getInput('env_vars'); // String of env vars KEY=VALUE,...
-    const secrets = core.getInput('secrets'); // String of secrets KEY=VALUE,...
-    const region = core.getInput('region') || 'us-central1';
-    const source = core.getInput('source'); // Source directory
-    const suffix = core.getInput('suffix');
-    const tag = core.getInput('tag');
-    const noTraffic = core.getBooleanInput('no_traffic');
-    const revTraffic = core.getInput('revision_traffic');
-    const tagTraffic = core.getInput('tag_traffic');
-    const flags = core.getInput('flags');
+    const inputs: DeployCloudRunInputs = {
```

Comment:
This is a pretty big change that might be better split up into a separate PR. It's also a bit of a departure from how our other actions establish state. I'm also a little worried about the values all becoming mutable now.

Reply:
Thought here was to enable future outputs depending on input variables (IE pass inputs object to parsing functions)

We can use Object.freeze or I can revert back and we can create an options parameter to the parsing functions.

Reply:
I'm not a huge fan of magic since it makes debugging more difficult. I think I'd rather revert this and have explicit inputs/outputs.

Reply:
Ack.